contracts/lens/Quoter.sol

```diff
@@ -0,0 +1,396 @@
+            sqrtPriceX96AfterList[i] = sqrtPriceX96After;
+            initializedTicksLoadedList[i] = initializedTicksLoaded;
+        }
+    }
+
+    function lockAcquired(bytes calldata encodedSwapIntention) external returns (bytes memory) {
+        if (msg.sender != address(manager)) {
+            revert InvalidLockAcquiredSender();
+        }
+
+        SwapInfo memory swapInfo = abi.decode(encodedSwapIntention, (SwapInfo));
+
+        if (swapInfo.swapType == SwapType.ExactInputSingle) {
+            (BalanceDelta deltas, uint160 sqrtPriceX96After, int24 tickAfter) =
+                _quoteExactInputSingle(abi.decode(swapInfo.params, (ExactInputSingleParams)));
+
+            bytes memory result = abi.encode(deltas, sqrtPriceX96After, tickAfter);
+            assembly {
+                revert(add(0x20, result), mload(result))
+            }
+        } else if (swapInfo.swapType == SwapType.ExactOutputSingle) {
+            (BalanceDelta deltas, uint160 sqrtPriceX96After, int24 tickAfter) =
+                _quoteExactOutputSingle(abi.decode(swapInfo.params, (ExactOutputSingleParams)));
+
+            bytes memory result = abi.encode(deltas, sqrtPriceX96After, tickAfter);
+            assembly {
+                revert(add(0x20, result), mload(result))
+            }
+        } else if (swapInfo.swapType == SwapType.ExactInput) {
+            (
+                int128[] memory deltaAmounts,
+                uint160[] memory sqrtPriceX96AfterList,
+                uint32[] memory initializedTicksLoadedList
+            ) = _quoteExactInput(abi.decode(swapInfo.params, (ExactInputParams)));
+
+            bytes memory result = abi.encode(deltaAmounts, sqrtPriceX96AfterList, initializedTicksLoadedList);
+            assembly {
+                revert(add(0x20, result), mload(result))
+            }
+        } else if (swapInfo.swapType == SwapType.ExactOutput) {
+            (
+                int128[] memory deltaAmounts,
+                uint160[] memory sqrtPriceX96AfterList,
+                uint32[] memory initializedTicksLoadedList
+            ) = _quoteExactOutput(abi.decode(swapInfo.params, (ExactOutputParams)));
+
+            bytes memory result = abi.encode(deltaAmounts, sqrtPriceX96AfterList, initializedTicksLoadedList);
+            assembly {
+                revert(add(0x20, result), mload(result))
+            }
+        } else {
+            revert InvalidQuoteType();
+        }
+    }
+
+    function validateRevertReason(bytes memory reason) private pure returns (bytes memory) {
+        if (reason.length < 96) {
+            // function selector + length of bytes as uint256 + min length of revert reason padded to multiple of 32 bytes
+            if (reason.length < 68) {
+                revert UnexpectedRevertBytes();
+            }
+            assembly {
+                reason := add(reason, 0x04)
+            }
+            revert(abi.decode(reason, (string)));
+        }
```

Comment:
constants for the magic #s and better comments to explain where they come from would be great